pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -301,7 +301,11 @@ func NewPlugin() plugin.Plugin {
 // OnOutboundListener is called whenever a new outbound listener is added to the LDS output for a given service
 // Can be used to add additional filters on the outbound path
 func (Plugin) OnOutboundListener(in *plugin.InputParams, mutable *plugin.MutableObjects) error {
-	return nil
+	if in.Node.Type != model.Router {
+		return nil
+	}
+
+	return setupFilters(in, mutable)
```

Comment:
Will this break any existing usage?
Let's say we have service foo and bar. Assume we only have rbac policy that allows GET access to foo for identity bar.

Before this PR, the rbac filter is generated only in the inbound listener of service foo. foo is still able to access other as there is no rbac filer on its outbound listener list.

After this PR, the rbac filter is generated in both inbound listener list and outbound listener list of foo. This has an side effect that foo is unable to access anyone in the mesh as its identity is always foo, not bar.

When generating the outbound listener, can we check the node.type and only generate if it's gateway? So that we exclude the sidecar in this case.

Also we need to test this in real deployment and add e2e tests. We can add e2e tests later but at least a manual test should be done before merging.

Reply:
Also RBAC on outbound is very tricky - it doesn't really provide any security, but may impact performance.

Reply:
But since it's restricted to gateway - I think it is ok.

Reply:
I think this do provide more security when the egress gateway is the only entity that is allowed to talk to the external in the mesh, like it's enforced by underlying firewall rule. In this case, using RBAC could provide more control on the egress traffic. For example: https://preliminary.istio.io/docs/examples/advanced-gateways/egress-gateway/

Also the RBAC is not enabled by default so if the user doesn't enable it, there is no performance impact.

Reply:
So the `in.Node.Type != model.Router` will make sure it only apply to gateway right? If so I think we just need to test it. It shouldn't break any existing use case.